src/aws/flb_aws_imds.c

```diff
@@ -0,0 +1,340 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
```

Comment:
Much of this file is refactored from [fliter_aws/aws.c](https://github.com/fluent/fluent-bit/blob/master/plugins/filter_aws/aws.c). Changes include 
- token refresh on failure
- IMDSv1 and IMDSv2 detection
- use of the flb_aws_client
- state management in IMDS context rather than in a specialized use case context.

Reply:
Cool- just make sure you add testing the filter to your test cases to double check nothing was broken.

Reply:
As noted in my design document, wiring the IMDS struct to the aws filter is out of the scope of this feature since I was hoping to not retest the aws plugin or introduce bugs to the plugin.

## Compatibility
The IMDS struct is designed to be fully compatible with plugins/aws_filters/aws.c with only about 5 lines of code changes. Moreover, much of aws.c's code can be deleted. 

The 5 code changes in aws.c that need to take place are listed in my design document.

## Future work
I expect that in the future, as changes are needed in aws.c and aws.c is being actively tested again, we can wire up the IMDS struct to aws.c and delete all the duplicated code.